### PR TITLE
Updating API endpoint for Metrics

### PR DESCRIPTION
**Pull request by a-jean**
This pull request is the server code for pushing metrics data to the Librato API(for public dashboard project -- Rails Girls Summer of Code). I created a controller called Metrics to collect client metrics information. Afterwards, it will use the Metriks gem to generate a hash and increment it. Lastly, I created a metrics file within config/initializers/ to insert code that will push these metrics directly to the Librato API(again through the use of the Metriks gem).

### CODE REVIEW

**Comment by dwradcliffe** [comment]
What is this for?

**Comment by lynnco** [comment]
Broadly: this is the Rails Girls Summer of Code work on analytics

Specifically: @a-jean is going to work on a description

**Comment by a-jean** [comment]
Hi @dwradcliffe, I've updated my PR and included a description.

**Comment by sonalkr132** [comment]
Hi @a-jean your commits could use a little help.
49f7c87 and 62619d0 are same, 95ea3bb is empty commit and c69fb20 shouldn't exist. Don't worry, it won't take much effort :smile: 

Here are a few sources you might find useful:
* [How to Write a Git Commit Message](http://chris.beams.io/posts/git-commit/)
* [Git Interactive Rebase, Squash, Amend and Other Ways of Rewriting History](https://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history)
*  [Git Branching - Rebasing](https://git-scm.com/book/en/v2/Git-Branching-Rebasing)

If you need help, feel free to reach out on #rubygems on freenode or slack.

Good luck on your project :+1:

**Comment by dwradcliffe** [comment]
I would really rather use Datadog instead of Librato for this. We already have everything setup for statsd/Datadog. I'd rather not add all these extra gem dependencies either.

File: app/controllers/api/v2/metrics_controller.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class Api::V2::MetricsController < Api::BaseController
+  METRICS = %I[
+    bundler_version
+    rubygems_version
+    ruby_version
+    ruby_platform
+    bundler_command
+    ruby_engine
+    bundler_settings
+    ci_information
```
Comment (Speculate7) [inline app/controllers/api/v2/metrics_controller.rb]: #

**Comment by indirect** [comment]
@dwradcliffe we've been collecting this set of data from bundler-api and sending it to Librato using this set of gems for a few years now, so this is the setup that I can help mentor in my sleep. I'm okay with using statsd instead, but I'll need someone to show me 1) how to send metrics using the existing setup, 2) how to create similar/equivalent dashboards from the data, and 3) how to make API requests that return specific slices of the data so we can build a Ruby-specific dashboard later. 👍

**Comment by a-jean** [comment]
Thanks for clarifying everyone! 
@sonalkr132 besides updating the commit messages, is there anything else that needs to be done with this PR?

**Comment by indirect** [comment]
@a-jean hi Ashley! so far, this pull request doesn’t have the functionality that we discussed in our last meeting. If you could use more clarification, check with your coaches first and then let me know if they could use any information from me. Thanks!

**Comment by jgaskins** [comment]
@indirect Responding to your Slack message here to keep things in one spot.

> at a minimum, the controller needs to expose a single route that accepts POST requests, accepts a list of names and values, and then sends each of those names and values to the metrics system

I'm not sure I understand what she's missing out of these requirements.

- [Exposes a single route that accepts POST requests](https://github.com/a-jean/rubygems.org/blob/b1eef1fe248cf3dd9f2660d1e3d78f1679163b1e/config/routes.rb#L17)
- [Accepts a list of names and values](https://github.com/a-jean/rubygems.org/blob/b1eef1fe248cf3dd9f2660d1e3d78f1679163b1e/app/controllers/api/v2/metrics_controller.rb#L13-L15) — if we get `ruby_version=2.3.1` in the params, this code increments the `ruby/2.3.1` counter
- [Sends each of those names and values to the metrics system](https://github.com/a-jean/rubygems.org/blob/b1eef1fe248cf3dd9f2660d1e3d78f1679163b1e/config/initializers/metriks.rb#L1-L2)

Since Metriks tracks its own instances in an internal registry, she doesn't need to keep handles to them individually. And since Metriks reporters runs on their own using the stats from that Metriks registry, she doesn't need to do anything special with it, either.

This PR _is_ missing an HTTP response, but that's a one-liner. Actually, now that I think about it, it wouldn't even affect the outcome here because even though this controller action would return an error response, it would still have captured the metrics by the time that happens. :-)

Is there something I'm overlooking?

**Comment by indirect** [comment]
@jgaskins thanks for following up! Sorry my spec for this feature has been so brief. I'll elaborate some more, and if anything is still confusing, please let me know!

1. The names in this PR don't match the names [currently being generated by Bundler](https://github.com/bundler/bundler/blob/master/lib/bundler/fetcher.rb#L170-L195). That means that this PR will create completely new metrics in Librato, which isn't what we want. We want to keep the same metric names.

2. This controller code will increment every single metric in the list every single request regardless of whether or not any data was provided — that's not helpful. We only want to send data to Librato when data is actually provided as part of the request. For example, if `bundler_command` is not included in a request to this code, it will increment a metric named `bundler_command/`, because `nil.to_s` evaluates to an empty string. This PR should not increment keys if no data is provided.

3. The existing code ignores many values that are actively reported by Bundler today. For example, [this line](https://github.com/bundler/bundler/blob/master/lib/bundler/fetcher.rb#L183) creates a metric whose name depends on the Ruby interpreter. We want to record those metrics whose names we don't know in advance.

**Comment by jgaskins** [comment]
@indirect All good points. Thanks!

**Comment by sonalkr132** [comment]
See #1634

**Pull request closed without merging**